Primera_entrega

manucho1586 opened the pull request.

manucho1586 committed:
Primera_entrega

manucho1586 committed:
Update README.md

manucho1586 committed:
Update README.md

EFilomena commented:
Emanuel, si me agragás como Reviewer, puedo empezar a revisar el código (EFilomena).
Gracias!

alutenberg commented on README.md:
```diff
@@ -1,2 +1,14 @@
-Primera_entrega
+# **<u>Trabajo Práctico N° 1</u>**
+___
+Para este primer trabajo se tuvieron en cuenta los 3 primeros cápitulos del libro utilizado en el curso. Donde se abordaron temas como manejo entradas y salidas digitales, manejo del tiempo de ejecucion y retardos por software y señales analógicas. También otros puntos importantes como modularización y forma prolija de escribir el código fuente para que pueda ser más mantenible y reutilizable.
```
curso, donde (está mal ese punto)
También está mal el punto antes de "también otros puntos": así como está escrita la oración debería ser una coma ahí.

manucho1586 replied:
Perfecto Ariel, las correcciones con respecto a la redacción ya están hechas.

alutenberg replied:
Ok, veo que esto ya lo corregiste acá: https://github.com/manucho1586/Garcia_TP1/blob/Correcciones_2/README.md

alutenberg commented on README.md:
```diff
@@ -1,2 +1,14 @@
+___
+Para este primer trabajo se tuvieron en cuenta los 3 primeros cápitulos del libro utilizado en el curso. Donde se abordaron temas como manejo entradas y salidas digitales, manejo del tiempo de ejecucion y retardos por software y señales analógicas. También otros puntos importantes como modularización y forma prolija de escribir el código fuente para que pueda ser más mantenible y reutilizable.
+
+Para poder llevar a cabo el objetivo planteado en la consigna del trabajo práctico, se trabajó con una libreria para el sensor de temperatura y humedad DHT11.
```
Acá sería bueno que agregues un link para indicar qué librería es la que se usó.

manucho1586 replied:
Link agregado.

alutenberg replied:
Ok, veo que esto ya lo corregiste acá: https://github.com/manucho1586/Garcia_TP1/blob/Correcciones_2/README.md

alutenberg commented on README.md:
```diff
@@ -1,2 +1,14 @@
+
+Para poder llevar a cabo el objetivo planteado en la consigna del trabajo práctico, se trabajó con una libreria para el sensor de temperatura y humedad DHT11.
+
+<b>*Se pudo cumplir con el funcionamiento planteado, el cual era imprimir los valores de temperatura y humedad a través del monitor del puerto serie, comparar el valor de temperatura con un umbral establecido y si la temperatura era superior a dicho umbral se prendian las alarmas (LED indicador y buzzer) y el ventilador. En caso contrario, las alarmas y el ventilador se mantenian apagados y hay u ...
```
No me queda claro por qué en este párrafo usas negritas y por qué usas itálicas. Sugiero no usar ninguna de las dos.

manucho1586 replied:
Simplemente era para mostrar la conclusión y como modo de práctica de las herramientas del markdown, ya las he desactivado.

alutenberg replied:
Ok, veo que esto ya lo corregiste acá: https://github.com/manucho1586/Garcia_TP1/blob/Correcciones_2/README.md

alutenberg commented on README.md:
```diff
@@ -1,2 +1,14 @@
+
+**Captura del monitor de la comunicación serie:**  
+  
+![Captura](file:///D:/Curso%20de%20Sistemas%20Embebidos%20FIUBA/TP's/Resolución%20del%20TP/Captura.PNG)
```
Me parece que tenes un problema con este link, porque la imagen no se ve.

manucho1586 replied:
Este link era una captura de pantalla de los datos impresos en el monitor serie, sigo teniendo ese problema, ya que en el programa de markdown me muestra que si esta la captura pero cuando pego en la hoja del readme del repositorio me aparece como un enlace.

alutenberg replied:
Te pasa esto porque si te fijas el link que figura en la línea 11 es (file:///D:/Curso%...) y Github no tiene acceso desde la página web a tu PC. Tendrías que subir la imagen al repositorio y apuntar el link ahí.

alutenberg commented on main.cpp:
```diff
@@ -0,0 +1,107 @@
+{
+//***********Inicialización de comunicación serie***************************//
+
+    UnbufferedSerial pc(USBTX, USBRX);      //Configuración de comunicación serie a través de la UART.
```
Acá hay una mezcla de comentarios en inglés y en español que no es muy prolija. Unificar todo en un idioma u otro.

alutenberg replied:
Ok, veo que esto ya lo corregiste acá: https://github.com/manucho1586/Garcia_TP1/blob/Correcciones_2/main.cpp

alutenberg commented on main.cpp:
```diff
@@ -0,0 +1,107 @@
+
+//*********Declaración de variables e inicialización de entradas-salidas****//
+    char buffer[16];                        //Buffer donde se almacenará los bytes de datos de temperatura y humedad.
+    int s;                                  //Variable donde se guardan los datos traidos por la función "readData()".     
```
Creo que sería mejor ponerle un nombre más descriptivo a esta variable.

alutenberg replied:
Ok, veo que esto ya lo corregiste acá: https://github.com/manucho1586/Garcia_TP1/blob/Correcciones_2/main.cpp

alutenberg commented on main.cpp:
```diff
@@ -0,0 +1,107 @@
+///**********Presionado de botón**********************************///
+///**********Este algoritmo de efecto antirrebote*****************///
+
+        while(boton_buzzer==LOW)
```
Esta línea, así como la línea 80, no me parecen adecuadas porque son muy bloqueantes.
¿Qué pasa si se queda siempre en el while en uno y otro caso? ¿Se puede cumplir con el funcionamiento planteado, "imprimir los valores de temperatura y humedad a través del monitor del puerto serie, comparar el valor de temperatura con un umbral establecido y si la temperatura era superior a dicho umbral se prenden las alarmas (LED indicador y buzzer) y el ventilador?

manucho1586 replied:
Esto pude corregir, era una rutina que tenia para cosas muy simples, que en este caso no me servía ya que pude comprobar que no me imprimia los datos en pantalla hasta que no soltara el botón, es decir, muy bloqueante como me lo mencionaste.
Para solucionar esto instancie una variable de la clase "ticker", la cual llamaba periodicamente a una función que chequeaba si los botones estaban presionados.

alutenberg replied:
Ok, veo que esto ya lo corregiste acá: https://github.com/manucho1586/Garcia_TP1/blob/Correcciones_2/main.cpp

manucho1586 committed:
Update main.cpp

En este nuevo branch se hacen las correcciones pertinentes, las cuales cosistian en comentarios, renombramientos de variables con nombres más intuitivos y cambios en en el algoritmo de presionado de botones. Estos últimos se cambiaron de una forma bloqueante a una no bloqueante, mediante un temporizador que llama a una función dedicada al estado de los botones, este llamado se realiza cada un (1) milisegundo.

manucho1586 committed:
Update README.md

Actualización del texto descriptivo (readme).

manucho1586 commented:
En este nuevo branch se hacen las correcciones pertinentes, las cuales cosistian en comentarios, renombramientos de variables con nombres más intuitivos y cambios en en el algoritmo de presionado de botones. Estos últimos se cambiaron de una forma bloqueante a una no bloqueante, mediante un temporizador que llama a una función dedicada al estado de los botones, este llamado se realiza cada un (1) milisegundo.

alutenberg reviewed (approved):
Queda aprobado tu TP1, sugiero que avances con el TP2.
Por favor fijate el correo con asunto "Ya pueden comenzar con el TP2" que envié el 
mié, 10 jul, 21:20 para ver todas las pautas de trabajo a aplicar. Cualquier duda me avisas.
